base/reduce.jl

```diff
@@ -228,9 +228,9 @@ with reduction `op` over an empty array with element type of `T`.
 If not defined, this will throw an `ArgumentError`.
 """
 reduce_empty(op, T) = _empty_reduce_error()
-reduce_empty(::typeof(+), T) = zero(T)
+reduce_empty(::typeof(+), ::Type{<:Number}) = zero(T)
```

Comment:
I've had to add this to make `sum` of an unknown iterator throw an `ArgumentError` instead of a `MethodError` on `zero(Any)`.

The benefit is that this is now consistent with `sum(Any[])` (which used to throw a `MethodError`).